src/main/scala/it/pagopa/interop/authorizationprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -0,0 +1,71 @@
+package it.pagopa.interop.authorizationprocess.common.readmodel
+
+import it.pagopa.interop.commons.cqrs.service.ReadModelService
+import it.pagopa.interop.authorizationmanagement.model.persistence.JsonFormats._
+import it.pagopa.interop.authorizationmanagement.model.client.PersistentClient
+import org.mongodb.scala.Document
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates.{`match`, count, project, sort}
+import org.mongodb.scala.model.Filters
+import org.mongodb.scala.model.Projections.{computed, fields, include}
+import org.mongodb.scala.model.Sorts.ascending
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ReadModelQueries {
+
+  def listClients(
+    name: Option[String],
+    relationshipIds: List[String],
+    consumerId: String,
+    purposeId: Option[String],
+    kind: Option[String],
```

Comment:
let's keep them typed
```suggestion
    relationshipIds: List[UUID],
    consumerId: UUID,
    purposeId: Option[UUID],
    kind: Option[PersistentClientKind],
```